src/main/c/Makefile

```diff
@@ -2,29 +2,84 @@
 
 # If you want to add a new cpp file, like foo.cpp, to the library, add foo.o to
 # this list
-OBJECTS=ibs.o davies.o
-
-CXX ?= c++
-# append existing flags so they override our flags
-CXXFLAGS += -O3 -march=native -g -std=c++11 -Ilibsimdpp-2.0-rc2 -Wall -Werror
-CXXFLAGS += -fPIC -ggdb
-LIBFLAGS += -fvisibility=hidden
+OBJECTS := \
+  ibs.o davies.o \
+  NativeModule.o NativePtr.o \
+  NativeStatus.o NativeLongFunc.o NativeCodeSuite.o
 
 UNAME_S := $(shell uname -s)
 UNAME_P := $(shell uname -p)
+
+# Compile with new version of clang if it is installed
+ifeq ($(UNAME_S),Linux)
+  ifndef CLANG_HOME
+    CLANG_HOME := $(shell /bin/ls -d /usr/l*/llvm* /usr/l*/llvm-5* /usr/l*/llvm-6* 2>/dev/null | tail -1)
+  endif
+  ifndef JAVA_HOME
+    JAVA_HOME := $(shell /bin/ls -d /usr/lib/jvm/java-8-*64 /usr/lib/jvm/java-9-*64 2>/dev/null | tail -1)
+  endif
+  JAVA_MD := linux
+else
+  CLANG_HOME := /usr/local/llvm-$(CLANG_VERSION)
+  ifndef JAVA_HOME
+    JAVA_JDK := $(shell /bin/ls -d /Library/Java/JavaVirtualMachines/jdk1.8*.jdk 2> /dev/null | tail -1)
+    ifneq ($(JAVA_JDK),)
+      JAVA_HOME := $(JAVA_JDK)/Contents/Home
+    endif
+  endif
+  JAVA_MD := darwin
+endif
+
+ifeq ($(CLANG_HOME),)
+  CLANG_HOME := CLANG_HOME_undefined
```

Comment:
Trying to ensure that if the configuration is bad, we get an error which makes it clear what was bad.  Otherwise an undefined variable gets treated as an empty string, which produces some bad filename without it being obvious what went wrong.